generator/processor.go

```diff
@@ -459,14 +459,29 @@ func joinDirectory(directory string, files []string) []string {
 	return result
 }
 
-var goPath = os.Getenv("GOPATH")
+var goPaths = filepath.SplitList(build.Default.GOPATH)
+
+func getPkgSrcDir(importPath string) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	pkg, err := build.Import(importPath, wd, 0)
+	if err != nil {
+		return "", err
+	}
+	return pkg.Dir, nil
+}
 
 func typeName(typ types.Type) string {
 	return removeGoPath(typ.String())
 }
 
 func removeGoPath(path string) string {
-	return strings.Replace(path, goPath+"/src/", "", -1)
+	for _, goPath := range goPaths {
```

Comment:
Might be better to use a [strings.Replacer](https://golang.org/pkg/strings/#Replacer) (though [strings.NewReplacer](https://golang.org/pkg/strings/#NewReplacer) is not very practical to use in this case)